component/client-handler/src/main/java/org/wso2/carbon/identity/oauth2/token/handler/clientauth/jwt/validator/JWTValidator.java

```diff
@@ -399,8 +399,9 @@ private boolean isValidSignature(String clientId, SignedJWT signedJWT, String te
                     isValidSignature = new JWKSBasedJWTValidator().validateSignature(jwtString, jwksUri, alg, options);
                 }
             } catch (IdentityOAuth2Exception e) {
-                String errorMessage = "Error occurred while validating signature using jwks";
-                throw new OAuthClientAuthnException(errorMessage, OAuth2ErrorCodes.INVALID_REQUEST, e);
+                String errorMessage = "Error occurred while validating signature using jwks ";
+                log.error(errorMessage, e);
+                return false;
```

Comment:
Without returning false lets throw the exception from this level. throw new OAuthClientAuthnException(errorMessage, OAuth2ErrorCodes.INVALID_CLIENT, e);

Reply:
if we throw the exception, the correct error code is not sent. That is why this was changed